model-engine/model_engine_server/infra/gateways/resources/k8s_endpoint_resource_delegate.py

```diff
@@ -1489,6 +1494,33 @@ async def _delete_keda_scaled_object(endpoint_id: str) -> bool:
                 return False
         return True
 
+    @staticmethod
+    async def _restart_deployment(deployment_name: str) -> None:
+        apps_client = get_kubernetes_apps_client()
+        # Patch the deployment with a timestamp label to restart it
+        patch = {
+            "spec": {
+                "template": {
+                    "metadata": {
+                        "annotations": {
+                            "model-engine.scale.com/restartedAt": datetime.datetime.now().isoformat()
+                        }
+                    }
+                }
+            }
+        }
+
+        try:
```

Comment:
should we be more explicit in the LeaderWorkerSet case? e.g. return that "it doesn't work for LWSs"

Reply:
make sense. let me do this in the use case code